pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,726 @@
+
+    /// Configure the pallet by specifying the parameters and types on which it depends.
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Because this pallet emits events, it depends on the runtime's definition of an event.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// Caps Currency
+        type Currency: Currency<Self::AccountId>;
+        /// Get information on nfts
+        type NFTHandler: NFTTrait<AccountId = Self::AccountId>;
+        /// Get information on marketplace
+        type MarketplaceHandler: MarketplaceTrait<Self::AccountId>;
+        /// Minimum required length of auction
+        #[pallet::constant]
+        type MinAuctionDuration: Get<Self::BlockNumber>;
+        /// Maximum permitted length of auction
+        #[pallet::constant]
+        type MaxAuctionDuration: Get<Self::BlockNumber>;
+        /// Minimum buffer of blocks required before auction start
+        #[pallet::constant]
+        type MinAuctionBuffer: Get<Self::BlockNumber>;
+        /// Grace period to extend auction by if new bid received
+        #[pallet::constant]
+        type AuctionGracePeriod: Get<Self::BlockNumber>;
+        /// Ending period during which an auction can be extended
+        #[pallet::constant]
+        type AuctionEndingPeriod: Get<Self::BlockNumber>;
+        /// The auctions pallet id - will be used to generate account id
+        #[pallet::constant]
+        type PalletId: Get<PalletId>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    // The pallet's runtime storage items.
+    // https://docs.substrate.io/v3/runtime/storage
+    #[pallet::storage]
+    #[pallet::getter(fn auctions)]
+    pub type Auctions<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        NFTId,
+        AuctionData<T::AccountId, T::BlockNumber, BalanceOf<T>>,
+        OptionQuery,
+    >;
+
+    // Pallets use events to inform users when important changes are made.
+    // https://docs.substrate.io/v3/runtime/events-and-errors
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// A new auction was created
+        AuctionCreated {
+            nft_id: NFTId,
+            marketplace_id: MarketplaceId,
+            creator: T::AccountId,
+        },
+        /// An existing auction was cancelled
+        AuctionCancelled {
+            nft_id: NFTId,
+            marketplace_id: MarketplaceId,
+            creator: T::AccountId,
+        },
+        /// An existing auction was purchased at buy it now price
+        AuctionBuyItNow {
+            nft_id: NFTId,
+            marketplace_id: MarketplaceId,
+            creator: T::AccountId,
+        },
+        /// An auction has completed and no more bids can be placed
+        AuctionCompleted {
```

Comment:
marketplace_id and creator is unnecessary. You are missing Who got it and for how much.